src/main/java/epicsquid/roots/integration/crafttweaker/tweaks/FlowerTweaker.java

```diff
@@ -81,7 +86,28 @@ public static void addRecipeItem(String name, IItemStack stack) {
       return;
     }
     Block block = ((ItemBlock) converted.getItem()).getBlock();
-    CraftTweaker.LATE_ACTIONS.add(new FlowerBlockMeta(name, block, converted.getMetadata()));
+    CraftTweaker.LATE_ACTIONS.add(new FlowerBlockMeta(name, block, converted.getMetadata(), null));
+  }
+
+  @ZenDocMethod(
+      order = 4,
+      args = {
+          @ZenDocArg(arg = "name", info = "The name of the recipe that you're adding"),
+          @ZenDocArg(arg = "stack", info = "The itemstack describing an itemblock to be placed"),
+          @ZenDocArg(arg = "allowedSoils", info = "A list of blocks that this flower can be placed on")
+      },
+      description = "Adds a recipe by creating a blockstate from an itemstack containing an itemblock and metadata to be grown during the Flower Growth ritual. The flower will only be grown on a soil mentioned. Accepts oredict entries."
+  )
+  @ZenMethod
+  public static void addRecipeItemOnSoils(String name, IItemStack stack, List<IIngredient> allowedSoils) {
+    ItemStack converted = CraftTweakerMC.getItemStack(stack);
+    if (!(converted.getItem() instanceof ItemBlock)) {
+      CraftTweakerAPI.logError("Cannot set " + stack.toString() + " as a Flower Growth ritual item as it is not an ItemBlock.");
+      return;
+    }
+    Block block = ((ItemBlock) converted.getItem()).getBlock();
+    CraftTweakerAPI.logDefault("addRecipeItemOnSoils with " + allowedSoils.size() + " soil types");
```

Comment:
If this is a debug, it should be removed. If it's not, it should be moved to `FlowerBlockMeta`, as otherwise the function fires the first time it's initialized (when the world starts), rather than when the actual recipe is created.